tuxemon/monster.py

```diff
@@ -442,29 +442,32 @@ def apply_status(self, status: Technique) -> None:
         if count_status == 0:
             self.status.append(status)
         else:
-            if self.status[0].category == "positive":
-                if status.repl_pos == "replace":
-                    self.status.insert(0, status)
-                    self.status.pop(1)
-                elif status.repl_pos == "remove":
-                    self.status.append(status)
-                    self.status.pop(0)
-                else:
-                    # noddingoff, exhausted, festering, dozing
-                    self.status.append(status)
-            elif self.status[0].category == "negative":
-                if status.repl_neg == "replace":
-                    self.status.insert(0, status)
-                    self.status.pop(1)
-                elif status.repl_pos == "remove":
-                    self.status.append(status)
-                    self.status.pop(0)
+            # if the status exists
+            if any(t for t in self.status if t.slug == status):
+                return
+            # if the status doesn't exist.
+            else:
+                if self.status[0].category == "positive":
+                    if status.repl_pos == "replace":
+                        self.status.clear()
+                        self.status.insert(0, status)
+                    elif status.repl_pos == "remove":
+                        self.status.clear()
+                    else:
+                        # noddingoff, exhausted, festering, dozing
+                        return
+                elif self.status[0].category == "negative":
+                    if status.repl_neg == "replace":
+                        self.status.clear()
+                        self.status.insert(0, status)
```

Comment:
I think "append" is more straightforward.

Reply:
It was append in the beginning, I can reinsert it eventually. I put insert because I wanted to reserve the second slot to spyderbite or eliminated, but as you said it can complicate the code.

So, this is it: self.status: List[Technique] = []

We get rid of the list and we replace it with? Str.enum? Other ideas?

Reply:
We need two slots: first for all the status, second for spyderbite or eliminated